src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3031,15 +3067,15 @@ void G1CollectedHeap::do_collection_pause_at_safepoint_helper(double target_paus
         // of the collection set!).
         _allocator->release_mutator_alloc_regions();
 
-        calculate_collection_set(evacuation_info, target_pause_time_ms);
+        calculate_collection_set(*jtm.evacuation_info(), target_pause_time_ms);
```

Comment:
`*jtm.evacuation_info()` seems odd; `calculate_collection_set` could use pointer instead of reference, right?

Reply:
It has been there that way, so I kept it in that change. Now changed all uses to pointers.